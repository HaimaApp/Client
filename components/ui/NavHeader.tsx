import { View, TouchableOpacity } from "react-native";
import React from "react";
import { FontAwesome5, Fontisto } from "@expo/vector-icons";
import { router } from "expo-router";

const NavHeader = () => {
  return (
    <View className="w-full px-5 flex-row items-center justify-end">
      {/* <View className="relative flex-row space-x-3 items-center justify-start">
        <View className="relative w-14 h-14 flex-row items-center justify-center bg-primary rounded-full">
          <Image
            source={icons.avatarIcon}
            alt="profile avatar"
            testID="profile-avatar"
          />
          <View className="w-4 h-4 bg-[#04802E] rounded-full border border-white absolute right-1 bottom-[1px]" />
        </View>
        <View className="flex-col items-start justify-start">
          <Text className="text-lg font-normal text-black">Aisha Uthman</Text>
          <Text className="text-xs font-medium text-black">Aisha Ventures</Text>
        </View>
      </View> */}
      <View className="flex-row items-center justify-end space-x-6">
        <TouchableOpacity testID="heart-button" onPress={() => router.push("/myfavourites")}>
          <FontAwesome5 name="heart" size={24} color="black" />
        </TouchableOpacity>
        <TouchableOpacity testID="bell-button" onPress={() => router.push("/notification/notify")}>
          <Fontisto name="bell" size={24} color="black" />
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default NavHeader;
