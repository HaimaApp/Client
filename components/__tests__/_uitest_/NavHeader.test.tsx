import NavHeader from "@/components/ui/NavHeader";
import renderer from "react-test-renderer";
import { router } from "expo-router";

jest.mock("expo-router", () => ({
  router: {
    push: jest.fn(),
  },
}));

jest.mock("@/constants", () => ({
  icons: {
    avatarIcon: "@/assets/icons/avatarIcon.png",
  },
}));

describe("<NavHeader />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<NavHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has the correct number of children", () => {
    const tree = renderer.create(<NavHeader />).toJSON();

    // Ensure tree is not null and is a single object (not an array)
    if (tree && !Array.isArray(tree) && "children" in tree) {
      expect(tree.children?.length).toBe(2); // Adjust the expected number based on your component's structure
    } else {
      // Fail the test if the structure is unexpected
      throw new Error("Tree structure is not as expected");
    }
  });
});
