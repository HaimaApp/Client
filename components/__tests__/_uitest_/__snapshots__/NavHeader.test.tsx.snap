// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavHeader /> renders correctly 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
      {
        "paddingLeft": 20,
        "paddingRight": 20,
      },
      {
        "flexDirection": "row",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "position": "relative",
        },
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "position": "relative",
          },
          {
            "width": 56,
          },
          {
            "height": 56,
          },
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "justifyContent": "center",
          },
          {
            "backgroundColor": "#FFCCCC",
          },
          {
            "borderBottomLeftRadius": 9999,
            "borderBottomRightRadius": 9999,
            "borderTopLeftRadius": 9999,
            "borderTopRightRadius": 9999,
          },
        ]
      }
    >
      <Image
        alt="profile avatar"
        source="@/assets/icons/avatarIcon.png"
      />
      <View
        style={
          [
            {
              "width": 16,
            },
            {
              "height": 16,
            },
            {
              "backgroundColor": "#04802E",
            },
            {
              "borderBottomLeftRadius": 9999,
              "borderBottomRightRadius": 9999,
              "borderTopLeftRadius": 9999,
              "borderTopRightRadius": 9999,
            },
            {
              "borderBottomWidth": 1,
              "borderLeftWidth": 1,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            {
              "borderBottomColor": "#fff",
              "borderLeftColor": "#fff",
              "borderRightColor": "#fff",
              "borderTopColor": "#fff",
            },
            {
              "position": "absolute",
            },
            {
              "right": 4,
            },
            {
              "bottom": 1,
            },
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "alignItems": "flex-start",
          },
          {
            "justifyContent": "flex-start",
          },
          [
            {
              "marginLeft": 12,
            },
          ],
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
            {
              "fontWeight": "400",
            },
            {
              "color": "#000",
            },
          ]
        }
      >
        Aisha Uthman
      </Text>
      <Text
        style={
          [
            {
              "fontSize": 12,
              "lineHeight": 16,
            },
            {
              "fontWeight": "500",
            },
            {
              "color": "#000",
            },
          ]
        }
      >
        Aisha Ventures
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": 24,
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
